sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -196,14 +194,8 @@ private[hive] object HiveShim {
       if (instance != null) {
         instance.asInstanceOf[UDFType]
       } else {
-        val func = Utils.getContextOrSparkClassLoader
+        Utils.getContextOrSparkClassLoader
           .loadClass(functionClassName).newInstance.asInstanceOf[UDFType]
-        if (!func.isInstanceOf[UDF]) {
```

Comment:
Doing a bit more research:

It looks like #3640 added this driver-side creation and serialization of the UDF not as a performance optimization but as a correctness issue. Given this, will this PR break the correctness for those UDFs?